### PR TITLE
FS-1760: Translations for magic-links

### PR DESCRIPTION
**Pull request by tferns**
**Summary:**
- Add remainder of translation tags for magic-links
- Add `-k lazy_gettext` to `README.md` on pybabel extraction

Superseded https://github.com/communitiesuk/funding-service-design-authenticator/pull/94 

**Note:**  Some translations are missing, we could attempt to translate these ourselves but they may not be exact - typically I'd just do this and then "ask for forgiveness" if they're incorrect and need changed, but I'm not too familiar with how appropriate that would be.   I'm going to raise that they're missing and hopefully it'll just be a case of editing the `.po` file & re-compiling once we have the appropriate translations.

### CODE REVIEW

File: frontend/translations/cy/LC_MESSAGES/messages.po [hunk 1]
```diff
@@ -23,10 +23,113 @@ msgstr ""
...
+msgid "Email not sent"
+msgstr ""
+
+#: frontend/magic_links/templates/check_email.html:14
+msgid "No email has been sent."
+msgstr ""
+
+#: frontend/magic_links/templates/check_email.html:15
+msgid "If you need a link to access your applications you can "
+msgstr ""
```
Comment (tferns) [inline frontend/translations/cy/LC_MESSAGES/messages.po:60]: Couldn't find these on the spreadsheet.

File: frontend/translations/cy/LC_MESSAGES/messages.po [hunk 2]
```diff
@@ -146,6 +245,38 @@ msgstr ""
...
+msgid "expired"
+msgstr ""
+
+#: frontend/magic_links/templates/signed_out.html:21
+msgid "been closed"
+msgstr ""
+
+#: frontend/magic_links/templates/signed_out.html:26
+msgid "To re-access your account, please click below to request a new link."
+msgstr ""
```
Comment (tferns) [inline frontend/translations/cy/LC_MESSAGES/messages.po:278]: More missing translations that I will raise.

File: frontend/magic_links/templates/magic_link.html [hunk 3]
```diff
@@ -5,8 +5,10 @@
...
 {{ govukPanel({
-  "titleText": "We've emailed you a link to sign in",
-  "html": "Check your email and follow the link to continue your application"
+"titleText": gettext("We've emailed you a link to sign in"),
+"html": gettext("Check your email and follow the link to continue your application")
 }) }}
 
 {% endblock %}
+
+{# TODO: is this being used? #}
```
Comment (tferns) [inline frontend/magic_links/templates/magic_link.html:14]: I couldn't find any usages of this macro

File: frontend/magic_links/templates/invalid.html [hunk 4]
```diff
@@ -4,23 +4,28 @@
...
-      <h1 class="govuk-heading-xl">Link expired</h1>
+      {# TODO: Figure out translation for Link expired #}
+      <h1 class="govuk-heading-xl">{% trans %} Link expired {% endtrans %}</h1>
   </div>
 <div class="govuk-error-summary" aria-labelledby="error-summary-title" role="alert" data-module="govuk-error-summary">
   <h2 class="govuk-error-summary__title" id="error-summary-title">
-      This link has expired
+      {# TODO: Figure out which should be used. #}
+      {# This link has expired #}
+      {% trans %} Magic link expired {% endtrans %}
```
Comment (tferns) [inline frontend/magic_links/templates/invalid.html:14]: `Magic link expired` was the closest translation I could find... so I replaced it and translated it.  I'll also raise this as I'm uncertain if this should've actually been replaced.

File: frontend/magic_links/templates/invalid.html [hunk 5]
```diff
@@ -4,23 +4,28 @@
...
   </h2>
   <div class="govuk-error-summary__body">
       <ul class="govuk-list govuk-error-summary__list">
       <li>
-        <p class="fsd-error-message">To access your application, you need to request a new link. The link will work for 24 hours.</p>
+        {# TODO: Figure out which should be used. #}
+        {# <p class="fsd-error-message">To access your application, you need to request a new link. The link will work for 24 hours.</p> #}
+        <p class="fsd-error-message">{% trans %} You can request a new magic link to access your application. {% endtrans %}</p>
```
Comment (tferns) [inline frontend/magic_links/templates/invalid.html:21]: Same scenario here, two similar bits of text, one has a translation and one doesn't, I'll raise this too.

**Comment by gidsg** [comment]
@tferns I think our policy has been not to use Google Translate, so I would leave out ones we don't have translations for,

**Comment by tferns** [comment]
> @tferns I think our policy has been not to use Google Translate, so I would leave out ones we don't have translations for,

I appreciate this clarification, as I'm relatively new it's difficult to gauge what's acceptable and what's not, slowly figuring it out 😆

Sounds good, that's the state of the PR at the moment 👍

**Comment by tferns** [comment]
> Are we holding off on this merge until all translations are in rather than have multiple pr's?

I think we'll hold off until https://github.com/communitiesuk/funding-service-design-authenticator/pull/93/ is in merged and in `main` - then I'll merge `origin/main` into this & fix conflicts

In terms of the Welsh stuff overall, I think we are safe to merge translations since they won't be exposed unless the user manually changes the cookie in their browser, the feature to do so via the UI is currently commented out and will act as the "release" for the Welsh changes: https://github.com/communitiesuk/funding-service-design-frontend/pull/220#issuecomment-1319933885

**Comment by tferns** [comment]
> Are we holding off on this merge until all translations are in rather than have multiple pr's?

Oh, I actually misinterpreted this.  I was going to merge this to prevent bit rot, then just raise another quick PR with the missing translation changes when we finally have them, should hopefully just be a change to the `.po` and `.mo` files 👍

**Review by gidsg — APPROVED** [review]
LGTM 👍

**Pull request merged**